Source/JavaScriptCore/API/JSValue.mm

```diff
@@ -159,6 +159,50 @@ + (JSValue *)valueWithNewSymbolFromDescription:(NSString *)description inContext
     return [JSValue valueWithJSValueRef:JSValueMakeSymbol([context JSGlobalContextRef], string.get()) inContext:context];
 }
 
++ (JSValue *)valueWithNewBigIntFromString:(NSString *)string inContext:(JSContext *)context
+{
+    JSValueRef exception = nullptr;
+    JSValueRef bigInt = JSBigIntCreateWithString([context JSGlobalContextRef], OpaqueJSString::tryCreate(string).get(), &exception);
+    if (exception) {
+        [context notifyException:exception];
+        return [JSValue valueWithUndefinedInContext:context];
```

Comment:
Maybe we should return `nullptr` here? Returning `undefined` is a bit confusing. It's probably worth changing the `valueWithNewPromiseInContext` return value too. I don't think that's used enough to be a bin compatibly risk.

Reply:
Would that cause any issue for the existing `valueWithNewPromiseInContext` applications since it's public?

Reply:
It's possible but unlikely since we return nullptr in other places the most consistent way to check for exceptions is to check the context.